### ISSUE TITLE
Missing commandline support - hacked in suggestion

### ISSUE THREAD

**Issue by marcakafoddex** [floooh/qoiview#13]
Commandline support would make this app even more useful, as you could register it as a viewer for .qoi files and have it display all files from the commandline.

I currently hackfixed this in (Windows support only) by doing:

in `typedef enum sapp_event_type {` I added:
```c
	SAPP_EVENTTYPE_INITIALIZED,
```

in `static void event(const sapp_event* ev) {` I added:

```c
	case SAPP_EVENTTYPE_INITIALIZED:
	for (int i = 0; i < _sargs.num_args; ++i) {
		const char* path = &_sargs.buf[_sargs.args[i].key];
		sfetch_send(&(sfetch_request_t) {
			.path = path,
				.callback = load_callback,
				.buffer = SFETCH_RANGE(state.file.buf),
		});
	}
	break;
```

your code does something special with a file path like `c:\foo\bar.qoi`, it turns it into `c:foobar.qoi`, so to turn `\` into `/`, I added in `sapp_desc sokol_main(int argc, char* argv[]) {` before `sargs_setup` is called:
```c
	for (int i = 0; i < argc; ++i) {
		char* argvch = argv[i];
		while (*argvch) {
			if (*argvch == '\\')
				*argvch = '/';
			++argvch;
		}
	}
```

and finally in `_SOKOL_PRIVATE void _sapp_win32_run(const sapp_desc* desc) {` I added:
```c
	bool initFired = false;
	...
	while (!(done || _sapp.quit_ordered)) {
		...
		if (!initFired) {
			initFired = true;
			_sapp_win32_app_event(SAPP_EVENTTYPE_INITIALIZED);
		}
	}
```

I don't know your sokol framework, so there might be a much better way of doing this, but this at least works, now i can register `qoiview` as a viewer for `.qoi` files in Windows :)

**Comment by floooh** [comment]
Hmm, I don't think changes to sokol_app.h (or an INITIALIZED event) are needed. The sfetch_send() calls would simply happen at the end of the init function:

https://github.com/floooh/qoiview/blob/2b064022730104850cba42eb8d26e83b8eb31719/qoiview.c#L238

sokol_args.h also recently got support for 'keyless args', check the `sargs_num_args()` and `sargs_value_at()` functions:

https://github.com/floooh/sokol/blob/eaa1ca79a4004750e58cb51e0100d27f23e3e1ff/sokol_args.h#L354-L359

...that way you don't need to access private state in the sokol_args.h implementation.

Loading multiple files wouldn't work with the rest of the code though, currently only one image is supported to be displayed.

I'm not sure why the backslashes are removed, as far as I can see there is no printf-style formatting happening anywhere in sokol_app.h, sokol_args.h or qoiview.c

**Comment by marcakafoddex** [comment]
You're right, adding it to the end of that init function works too, i completely missed that one. But then again, I was just looking to make it work quick and dirty :D anyway, probably as the author you're the best person to implement this anyway ;-)
